snakemake/executors/__init__.py

```diff
@@ -33,6 +33,7 @@
 from snakemake.logging import logger
 from snakemake.stats import Stats
 from snakemake.utils import format, Unformattable, makedirs
+from snakemake.provenance_tracking.provenance import provenance_manager
```

Comment:
Since the provenance manager only has to be part of AbstractExecutor, I think we could avoid the singleton and just keep it in there instead.